Add names to repr for named tensors

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23625 Implement tensor.align_to(names), torch.align_tensors(*tensors)
* #23624 Add name propagation for at::alias, add tensor.set_names
* **#23316 Add names to repr for named tensors**
* #23623 Add torch._C._BUILD_NAMEDTENSOR()

Adds a `names=` argument to the repr() of tensor if `tensor.has_names()`.

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 23316 gh/zou3519/80/head

Differential Revision: [D16494415](https://our.internmc.facebook.com/intern/diff/D16494415)

zou3519 committed:
Implement repr for named tensors

Adds a `names=` argument to the repr() of tensor if `tensor.has_names()`.

Test Plan:
- [namedtensor ci]

nairbv commented on test/test_namedtensor.py:
```diff
@@ -85,6 +85,18 @@ def test_has_names(self):
         self.assertTrue(fully_named.has_names())
 
+    def test_repr(self):
+        named_tensor = torch.zeros(2, 3).set_names_(['N', 'C'])
```
It's not really related to this particular PR, but do we have other tensor functions named this way? To me the `set` and trailing `_` seem to redundantly express the same thing. I feel like we could have a better name for it.

zou3519 replied:
We have:
- set_data(...)
- set_
- set_quantizer_
so an example and a counter-example. I'm taking suggestions for a better name!

nairbv commented:
I think it'd be clearer to name this commit message something like 'add names to tensor repr,' I got a bit confused thinking I'd see some newly implemented repr() function.

zou3519 committed:
Update on "Add names to repr for named tensors"

Add names to repr for named tensors

Adds a `names=` argument to the repr() of tensor if `tensor.has_names()`.

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 23316 gh/zou3519/80/head

zou3519 committed:
Update on "Add names to repr for named tensors"

Add names to repr for named tensors

Adds a `names=` argument to the repr() of tensor if `tensor.has_names()`.

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 23316 gh/zou3519/80/head

zou3519 commented:
@pytorchbot retest this please

zou3519 committed:
Update on "Add names to repr for named tensors"

Add names to repr for named tensors

Adds a `names=` argument to the repr() of tensor if `tensor.has_names()`.

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 23316 gh/zou3519/80/head

zou3519 committed:
Update on "Add names to repr for named tensors"

Add names to repr for named tensors

Adds a `names=` argument to the repr() of tensor if `tensor.has_names()`.

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 23316 gh/zou3519/80/head

zou3519 committed:
Update on "Add names to repr for named tensors"

Add names to repr for named tensors

Adds a `names=` argument to the repr() of tensor if `tensor.has_names()`.

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 23316 gh/zou3519/80/head

zou3519 committed:
Update on "Add names to repr for named tensors"

Add names to repr for named tensors

Adds a `names=` argument to the repr() of tensor if `tensor.has_names()`.

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 23316 gh/zou3519/80/head

zou3519 committed:
Update on "Add names to repr for named tensors"

Add names to repr for named tensors

Adds a `names=` argument to the repr() of tensor if `tensor.has_names()`.

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 23316 gh/zou3519/80/head

zou3519 committed:
Update on "Add names to repr for named tensors"

Add names to repr for named tensors

Adds a `names=` argument to the repr() of tensor if `tensor.has_names()`.

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 23316 gh/zou3519/80/head
